proparse/src/main/java/org/prorefactor/treeparser01/TP01Support.java

```diff
@@ -842,14 +868,14 @@ public void paramForRoutine(AST directionAST) {
    * @param datatypeAST The node of the datatype, might be a CLASS node.
    */
   @Override
-  public void paramNoName(AST datatypeAST) {
+  public void paramNoName(JPNode typeNode) {
+    LOG.trace("Entering paramNoName {}", typeNode);
     Variable variable = new Variable("", currentScope);
     currSymbol = variable;
-    JPNode typeNode = (JPNode) datatypeAST;
-    if (typeNode.getType() == NodeTypes.CLASS)
+    if (typeNode.getType() == ProParserTokenTypes.CLASS)
       typeNode = typeNode.nextNode();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Introduce a new variable instead of reusing the parameter "typeNode". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS1226)